### ISSUE TITLE
Bug: [space-infix-ops] rule reports incorrect errors for type alias declaration

### ISSUE THREAD

**Issue by armano2** [typescript-eslint/typescript-eslint#5137]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&showAST=es&code=C4TwDgpgBAYg9nAPAFShAHsCA7AJgZygAoA6MgQwCcBzfALinOxAG0BdASigF4A+R5v25RkAbiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZYgQzMVuYDNKAD1oB7YqgyRE0aKOiRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type Foo<T extends (...args: any[]) => any> = T;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/space-infix-ops": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors

### Actual Result

> Operator ':' must be spaced. 1:28 - 1:29

### Additional Info

 https://github.com/typescript-eslint/typescript-eslint/pull/5113#issuecomment-1144990098

cc @pkudryashov

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.27.0` |
| `@typescript-eslint/parser` | `5.27.0` |
| `TypeScript` | `4.7.2` |
| `ESLint` | `8.15.0` |
| `node` | `web` |
